extensions/wikia/Search/classes/Hooks.php

```diff
@@ -3,161 +3,196 @@
+		return ( new Indexer )->reindexBatch( [ $title->getArticleID() ] );
 	}
-	
+
 	/**
 	 * Issues a reindex event or deletes all docs, depending on whether a wiki is being closed or reopened
-	 * @param  int    $city_public
-	 * @param  int    $city_id
+	 *
+	 * @param  int $city_public
+	 * @param  int $city_id
 	 * @param  string $reason
+	 *
 	 * @return bool
 	 */
 	public static function onWikiFactoryPublicStatusChange( &$city_public, &$city_id, $reason ) {
-		return ( $city_public < 1 ) 
-		    ? (new Indexer)->deleteWikiDocs( $city_id )
-		    : (new Indexer)->reindexWiki( $city_id );
+		return ( $city_public < 1 ) ? ( new Indexer )->deleteWikiDocs( $city_id ) :
+			( new Indexer )->reindexWiki( $city_id );
 	}
-	
+
 	/**
-	 * Used to configure the user preference pane settings for search. 
+	 * Used to configure the user preference pane settings for search.
 	 * This is a registered hook function of the samme name.
-	 * 
-	 * @param User $user
+	 *
+	 * @param \User $user
 	 * @param array $defaultPreferences
-	 */ 
+	 */
 	public static function onGetPreferences( $user, &$defaultPreferences ) {
 		// removes core mw search prefs
-		$defunctPreferences = array(
+		$defunctPreferences = [
 			'searchlimit',
 			'contextlines',
 			'contextchars',
 			'disablesuggest',
 			'searcheverything',
 			'searchnamespaces',
-		);
+		];
 
 		foreach ( $defunctPreferences as $goAway ) {
 			unset( $defaultPreferences[$goAway] );
 		}
 
-		$defaultPreferences["enableGoSearch"] = array(
-			'type'			=> 'toggle',
-			'label-message'	=> array('wikiasearch2-enable-go-search'),
-			'section'		=> 'under-the-hood/advanced-displayv2',
-		);
-
-		$defaultPreferences["searchAllNamespaces"] = array(
-			'type'			=> 'toggle',
-			'label-message'	=> array('wikiasearch2-search-all-namespaces'),
-			'section'		=> 'under-the-hood/advanced-displayv2',
-		);
+		$defaultPreferences["enableGoSearch"] = [
+			'type' => 'toggle',
+			'label-message' => [ 'wikiasearch2-enable-go-search' ],
+			'section' => 'under-the-hood/advanced-displayv2',
+		];
+
+		$defaultPreferences["searchAllNamespaces"] = [
+			'type' => 'toggle',
+			'label-message' => [ 'wikiasearch2-search-all-namespaces' ],
+			'section' => 'under-the-hood/advanced-displayv2',
+		];
+
 		return true;
 	}
-	
+
 	/**
 	 * WikiaMobile hook to add assets so they are minified and concatenated
+	 *
 	 * @see    SearchControllerTest::testOnWikiaMobileAssetsPackages
-	 * @param  array $jsHeadPackages
-	 * @param  array $jsBodyPackages
+	 *
+	 * @param  array $jsStaticPackages
+	 * @param  array $jsExtensionPackages
 	 * @param  array $scssPackages
+	 *
 	 * @return boolean
 	 */
-	public static function onWikiaMobileAssetsPackages( &$jsStaticPackages, &$jsExtensionPackages, &$scssPackages){
-		if( \F::app()->wg->Title->isSpecial( 'Search' ) ) {
+	public static function onWikiaMobileAssetsPackages( &$jsStaticPackages, &$jsExtensionPackages, &$scssPackages ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$jsStaticPackages". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)